bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -181,60 +126,40 @@ public BigtableOptions build() {
   private final InetAddress tableAdminHost;
   private final InetAddress dataHost;
   private final int port;
-  private final Credentials credential;
   private final String projectId;
   private final String zone;
   private final String cluster;
-  private final RetryOptions retryOptions;
-  private final String callTimingReportPath;
-  private final String callStatusReportPath;
-  private final ScheduledExecutorService rpcRetryExecutorService;
   private final EventLoopGroup customEventLoopGroup;
-  private final int channelCount;
-  private final long timeoutMs;
-  private final String userAgent;
+  private final ChannelOptions channelOptions;
 
   private BigtableOptions(
       InetAddress clusterAdminHost,
       InetAddress tableAdminHost,
       InetAddress dataHost,
       int port,
-      Credentials credential,
       String projectId,
       String zone,
       String cluster,
-      RetryOptions retryOptions,
-      String callTimingReportPath,
-      String callStatusReportPath,
-      ScheduledExecutorService rpcRetryExecutorService,
       EventLoopGroup customEventLoopGroup,
-      int channelCount,
-      long timeoutMs,
-      String userAgent) {
+      ChannelOptions channelOptions) {
     Preconditions.checkArgument(
         !Strings.isNullOrEmpty(projectId), "ProjectId must not be empty or null.");
     Preconditions.checkArgument(
         !Strings.isNullOrEmpty(zone), "Zone must not be empty or null.");
     Preconditions.checkArgument(
         !Strings.isNullOrEmpty(cluster), "Cluster must not be empty or null.");
-    Preconditions.checkArgument(
-      !Strings.isNullOrEmpty(userAgent), "UserAgent must not be empty or null.");
+    Preconditions.checkNotNull(channelOptions, "ChannelCoptions must not be empty or null.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(channelOptions.getUserAgent()),
+      "UserAgent must not be empty or null");
```

Comment:
Yes.  That's the way it's been since the user agent functionality was first committed.  See https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java#L270